src/trans.rs

```diff
@@ -72,6 +107,10 @@ impl<'v, 'tcx: 'v> Drop for BinaryenModuleCtxt<'v, 'tcx> {
     }
 }
 
+// The address in wasm linear memory where we store the stack pointer
+// TODO: investigate where should the preferred location be
+const STACK_POINTER_ADDRESS : i32 = 0;
```

Comment:
I realize that on IRC I kind of advocated for sticking the stack pointer at 0 just because we can and I think that's cool. From a software reliability standpoint, this is probably not the best idea. For pure Rust code, we're probably safe from accidentally dereferencing a null pointer, but if we ever link in something from C we could end up with a case where something that might be expected to crash would instead overwrite the stack pointer. This will probably lead to a crash later, so maybe it's not a big deal, but I could also see this opening up security problems (although the WASM sandbox will probably protect us somewhat here).

The nice thing about this being a constant is that it's trivial to change. We should keep an eye out for any unsafe pointer casts in real-world code (or as close as we have, anyway) that mir2wasm compiles, since these are where the risks will come from.